### PR TITLE
added Godoc to api types where necessary

### PR DESCRIPTION
**Pull request by stevekuznetsov**
@smarterclayton  PTAL

This pull is straight adds only.

### CODE REVIEW

File: pkg/authorization/api/v1/types.go [hunk 1]
```diff
@@ -42,7 +42,8 @@ type IsPersonalSubjectAccessReview struct {
 // Role is a logical grouping of PolicyRules that can be referenced as a unit by RoleBindings.
 type Role struct {
 	unversioned.TypeMeta `json:",inline"`
-	kapi.ObjectMeta      `json:"metadata,omitempty"`
```
Comment (stevekuznetsov) [inline pkg/authorization/api/v1/types.go]: @deads2k this is what I meant

File: pkg/image/api/v1/types.go [hunk 2]
```diff
@@ -201,19 +213,27 @@ type ImageStreamImage struct {
 
 // DockerImageReference points to a Docker image.
 type DockerImageReference struct {
```
Comment (stevekuznetsov) [inline pkg/image/api/v1/types.go]: Does this object need `json` struct tags? Do not ever serialize it? if not, what is it doing in this file?

Comment (smarterclayton) [inline pkg/image/api/v1/types.go]: Hrm, I can't remember if we needed it.

**Comment by smarterclayton** [comment]
[merge]

**Comment by smarterclayton** [comment]
[merge]

**Pull request merged**